connector/oidc/oidc.go

```diff
@@ -309,27 +369,30 @@ var (
 )
 
 type oidcConnector struct {
-	provider                  *oidc.Provider
-	redirectURI               string
-	oauth2Config              *oauth2.Config
-	verifier                  *oidc.IDTokenVerifier
-	cancel                    context.CancelFunc
-	logger                    *slog.Logger
-	httpClient                *http.Client
-	insecureSkipEmailVerified bool
-	insecureEnableGroups      bool
-	allowedGroups             []string
-	acrValues                 []string
-	getUserInfo               bool
-	promptType                string
-	userIDKey                 string
-	userNameKey               string
-	overrideClaimMapping      bool
-	preferredUsernameKey      string
-	emailKey                  string
-	groupsKey                 string
-	newGroupFromClaims        []NewGroupFromClaims
-	groupsFilter              *regexp.Regexp
+	provider                     *oidc.Provider
+	redirectURI                  string
+	oauth2Config                 *oauth2.Config
+	verifier                     *oidc.IDTokenVerifier
+	cancel                       context.CancelFunc
+	logger                       *slog.Logger
+	httpClient                   *http.Client
+	insecureSkipEmailVerified    bool
+	insecureEnableGroups         bool
+	allowedGroups                []string
+	acrValues                    []string
+	getUserInfo                  bool
+	promptType                   string
+	userIDKey                    string
+	userNameKey                  string
+	overrideClaimMapping         bool
+	preferredUsernameKey         string
+	emailKey                     string
+	groupsKey                    string
+	newGroupFromClaims           []NewGroupFromClaims
+	groupsFilter                 *regexp.Regexp
+	pkceChallenge                string
+	pkceMaxConcurrentConnections int
+	pkceVerifierCache            *lru.Cache[string, string]
```

Comment:
Hi @nabokihms, thanks for your answer.
So you recommend to use database?
If so, how could it be implemented?

Reply:
In the authrequest there is a field [`connectorData`](https://github.com/dexidp/dex/blob/2ab83e52d914e14554d91fae7f3ece9f1322afe9/storage/storage.go#L229).

I think we can use it between login and callback interactions with the connector. 
One way to achieve this is to change this interface (which was rock solid since 2016)
https://github.com/dexidp/dex/blob/2ab83e52d914e14554d91fae7f3ece9f1322afe9/connector/connector.go#L52-L70 

Another way is to give connectors access to the storage which is a no go scenario. 

So changing the interface is the only way, and I remember there was a PR that did exactly what we need here https://github.com/dexidp/dex/pull/2041 You may catch some ideas from here. 

And... we also need to add more tests to do such a change.

Reply:
I'm ready to merge it if you decide to dive into implementation.